examples/contrib/readhar.py

```diff
@@ -0,0 +1,134 @@
+"""Reads HAR files into flow objects"""
+import asyncio
+import codecs
+import json
+import logging
+import time
+
+from mitmproxy import command
+from mitmproxy import connection
+from mitmproxy import ctx
+from mitmproxy import exceptions
+from mitmproxy import http
+from mitmproxy import types
+
+
+class ReadHar:
+    def __init__(self):
+        self.harFile = {}
+        self.flows = []
+
+    def fix_headers(self, request_headers):
+        """Changes provided headers into flow-friendly format"""
+        flow_headers = []
+        for header in request_headers:
+            if type(header) == dict and header.get("name", None):
+                key = header["name"]
+                value = header["value"]
+            else:
+                key = header[0]
+                value = header[1]
+            flow_headers.append((bytes(key, "utf-8"), bytes(value, "utf-8")))
+
+        return flow_headers
+
+    def request_to_flow(self, request_json):
+        """
+        Creates a HTTPFlow object from a given entry in HAR file
+        """
+
+        requestMethod = request_json["request"].get("method", "")
+        requestUrl = request_json["request"].get("url", None)
+        if not requestUrl:
+            requestUrl = request_json["request"].get("host", None)
+
+        requestHeaders = self.fix_headers(request_json["request"].get("headers", []))
+
+        client_conn = connection.Client(
+            peername=("127.0.0.1", 51513),
+            sockname=("127.0.0.1", 8080),
+            timestamp_start=time.time(),
+        )
+        server_conn = connection.Server(address=None)
+
+        newflow = http.HTTPFlow(client_conn, server_conn)
+        try:
+            newflow.request = http.Request.make(
+                requestMethod, requestUrl, "", requestHeaders
+            )
+        except:
```

Comment:
Two notes on style:

1. Bare except statements are evil in Python, see https://www.flake8rules.com/rules/E722.html. If unsure about the error `except Exception` is better to avoid catching `KeyboardInterrupt`. See https://docs.python.org/3/library/exceptions.html#exception-hierarchy.